kubernetes/cray-tapms-crd/files/tapms.hpe.com_globaltenanthooks.yaml

```diff
@@ -0,0 +1,89 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.11.1
+  creationTimestamp: null
+  name: globaltenanthooks.tapms.hpe.com
+spec:
+  group: tapms.hpe.com
+  names:
+    kind: GlobalTenantHook
+    listKind: GlobalTenantHookList
+    plural: globaltenanthooks
+    singular: globaltenanthook
+  scope: Namespaced
+  versions:
+  - name: v1alpha3
+    schema:
+      openAPIV3Schema:
+        description: '@Description The primary schema/definition of a global tenant
+          hook'
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: The desired state of the global tenant webhook
+            properties:
+              blockingcall:
+                default: false
+                type: boolean
+              eventtypes:
+                items:
+                  type: string
+                type: array
+              hookcredentials:
+                description: '@Description Optional credentials for calling webhook'
+                properties:
+                  secretname:
+                    description: Optional Kubernetes secret name containing credentials
+                      for calling webhook
+                    type: string
+                  secretnamespace:
+                    description: Optional Kubernetes namespace for the secret
+                    type: string
+                type: object
+              name:
+                type: string
+              url:
+                type: string
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
```

Comment:
It appears this file is generated by the Makefile, but it is not quite identical?
```
diff config/crd/bases/tapms.hpe.com_globaltenanthooks.yaml kubernetes/cray-tapms-crd/files/tapms.hpe.com_globaltenanthooks.yaml
89c89
<       status: {}
---
>       status: {}
\ No newline at end of file
```
Please add the missing newline.

Reply:
I'm also curious, if this file is being generated by `make manifests`, why does it need to be checked in?  From the Makefile:
```
.PHONY: manifests
manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
        $(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
        cp ./config/crd/bases/tapms.hpe.com_tenants.yaml ./kubernetes/cray-tapms-crd/files/tapms.hpe.com_tenants.yaml
        cp ./config/crd/bases/tapms.hpe.com_globaltenanthooks.yaml ./kubernetes/cray-tapms-crd/files/tapms.hpe.com_globaltenanthooks.yaml
```

Reply:
I'm unsure.  I'm just following the process that Brad Klein laid out.  Given the time pressure to get this in I'm trying mess with as little as possible right now.